src/base/utils/misc.h

```diff
@@ -84,7 +84,7 @@ namespace Utils
 
         // Take a number of seconds and return a user-friendly
         // time duration like "1d 2h 10m".
-        QString userFriendlyDuration(qlonglong seconds);
+        QString userFriendlyDuration(qlonglong seconds, qlonglong maxCap = -1);
```

Comment:
Or maybe make "0 means unlimited" since 0 isn't appropriate maxCap value? (actually `maxCap <= 0` means "unlimited" in your code)

Reply:
Personally I like `-1` better, also I think capping at `0` is valid since that is a non-negative number (even if there is no point doing so). Will update it to make `0` a valid cap.